feat: added support for `uint64` keys

theanmolsharma opened the pull request:
In order to improve bcoin's coin selection and make it faster, we need to reorganize how we store UTXOs in `lib/wallet/txdb.js`. An obvious optimization is to store unspent credits separately and fetch only unspent coins. Currently, `getCredits()` in `txdb.js`, ignores all spent credits but has to iterate through all coins(both spent and unspent).

A more advanced optimization would be to select coins in some range. For example, if we want to spend 200 Satoshis, we can fetch all coins in the range of 100 to 300, for coin selection. In order to do so, we need to store coins sorted by `value`. Since bitcoin's output values are 64-bit integer values, they can not directly be stored because `bdb` doesn't support `uint64` values. 

A workaround is to use two `uint32` values. But in the long run, it will be better to add support for `uint64` as it may be required in the future for more advanced optimization.

theanmolsharma committed:
feat: added support for `uint64` keys

pinheadmz commented on .gitignore:
```diff
@@ -17,3 +17,4 @@ node_modules/
 npm-debug.log
 prebuilds/
 yarn.lock
+.idea
```
can we remove this? IDE-specific ignore files should go in a global gitignore on your system

theanmolsharma replied:
Fixed it. I think we can merge now!!

theanmolsharma replied:
Yes, I'll fix it.

pinheadmz commented:
@theanmolsharma I also added tests and pushed to your branch at https://github.com/bcoin-org/bdb/pull/14/commits/e6435f171e9965571e2f9eaef7d58f8c6240e4b9

Sorry I promise I won't do this a lot, but this package is missing test coverage and for my own sanity I wanted to make sure uint64 was safe to use. Take a look at the tests and let me know what you think. Otherwise, I think this is a good addition to bdb and we should merge it and release.

theanmolsharma commented:
> @theanmolsharma I also added tests and pushed to your branch at [e6435f1](https://github.com/bcoin-org/bdb/commit/e6435f171e9965571e2f9eaef7d58f8c6240e4b9)
> 
> Sorry I promise I won't do this a lot, but this package is missing test coverage and for my own sanity I wanted to make sure uint64 was safe to use. Take a look at the tests and let me know what you think. Otherwise, I think this is a good addition to bdb and we should merge it and release.



> @theanmolsharma I also added tests and pushed to your branch at [e6435f1](https://github.com/bcoin-org/bdb/commit/e6435f171e9965571e2f9eaef7d58f8c6240e4b9)
> 
> Sorry I promise I won't do this a lot, but this package is missing test coverage and for my own sanity I wanted to make sure uint64 was safe to use. Take a look at the tests and let me know what you think. Otherwise, I think this is a good addition to bdb and we should merge it and release.

It's okay that you added tests. I myself wanted to add tests but I wanted your review before doing that.
The tests look good. I think we can merge and release.

theanmolsharma commented:
> @theanmolsharma I also added tests and pushed to your branch at [e6435f1](https://github.com/bcoin-org/bdb/commit/e6435f171e9965571e2f9eaef7d58f8c6240e4b9)
> 
> Sorry I promise I won't do this a lot, but this package is missing test coverage and for my own sanity I wanted to make sure uint64 was safe to use. Take a look at the tests and let me know what you think. Otherwise, I think this is a good addition to bdb and we should merge it and release.



> @theanmolsharma I also added tests and pushed to your branch at [e6435f1](https://github.com/bcoin-org/bdb/commit/e6435f171e9965571e2f9eaef7d58f8c6240e4b9)
> 
> Sorry I promise I won't do this a lot, but this package is missing test coverage and for my own sanity I wanted to make sure uint64 was safe to use. Take a look at the tests and let me know what you think. Otherwise, I think this is a good addition to bdb and we should merge it and release.

It's okay that you added tests. I myself wanted to add tests but I wanted your review before doing that.
The tests look good. I think we can merge and release.

pinheadmz commented:
ok LGTM, lets get a second review from @nodech in case I'm missing something important
